lib/strava/api/pagination.rb

```diff
@@ -0,0 +1,43 @@
+module Strava
+  module Api
+    #
+    # Wrapper Class around pagination results which are an Array of Models.
+    #
+    # This class exists to provide a way to provide a high-level abstraction, like
+    # access to API request/reponse data (i.e. ratelimits) for paginated results.
+    #
+    class Pagination
+      include Enumerable
+
+      attr_reader :collection
+
+      def initialize(collection)
+        @collection = collection
+      end
+
+      def size
+        @collection.size
+      end
+
+      def each
+        @collection.each { |c| yield c if block_given? }
+      end
+
+      def ratelimit
+        return nil unless @collection.first
+
+        @collection.first.ratelimit
```

Comment:
The response may not have any items in it, but have rate limit information. So I think you can't lift it from the first element.

Btw, I'm ok deferring fixing this to later with an issue after we merge.

Reply:
You were absolutely right. Let's not haste and take the time to not have to work on it later.

I put some more work into it, because if we merge it, it's going to be around for some time. And personally I 
wasn't too happy with all the `method_missing` and the footprint this feature already brought with it.

So, I feel it's really important to keep the public facing API of this feature as small as possible. I will reread all variables once more, so request and response wasn't mixed up and have the YARD documentation matching.